frontend/src/modules/studySamples/services.ts

```diff
@@ -163,64 +129,61 @@ export async function buildStudySamplesFromWorkflow(
-			const user = process && process.created_by ? usersByID[process.created_by] : stepHistory.created_by ? usersByID[stepHistory.created_by] : undefined
-			
-			const completedSample : CompletedStudySample = {
-				id: stepHistory.id,
-				sampleID: stepHistory.sample,
-				generatedSampleID: processMeasurement?.child_sample,
-				processID: processMeasurement?.process,
-				processMeasurementID: stepHistory.process_measurement,
-				executionDate: processMeasurement ? processMeasurement?.execution_date : timestampStringAsDate(stepHistory.created_at),
-				executedBy: user?.username,
-				comment: processMeasurement?.comment,
-				removedFromWorkflow: stepHistory.workflow_action === 'DEQUEUE_SAMPLE'
-			}
-			step.completed.push(completedSample)
-		}
+	const completedStudySamples = completedSamples.map((completedSample) => {
+		const processMeasurement = processMeasurementsByID[completedSample.process_measurement]
+		const process = processMeasurement ? processesByID[processMeasurement.process] : undefined
+		const user = process && process.created_by ? usersByID[process.created_by] : completedSample.created_by ? usersByID[completedSample.created_by] : undefined
+
+		return {
+			id: completedSample.id,
+			sampleID: completedSample.sample,
+			generatedSampleID: processMeasurement?.child_sample,
+			processID: processMeasurement?.process,
+			processMeasurementID: completedSample.process_measurement,
+			executionDate: processMeasurement ? processMeasurement?.execution_date : timestampStringAsDate(completedSample.created_at),
+			executedBy: user?.username,
+			comment: processMeasurement?.comment,
+		} as CompletedStudySample
 	})
 
-	// Return the steps in the step order
-	const steps = Array.from(stepMap.values()).sort((a, b) => a.stepOrder - b.stepOrder)
-
 	return {
-		steps
+		result: completedStudySamples,
+		count: totalCount
 	}
 }
 
-export async function fetchSamplesAtStepOrder(studyID: FMSId, stepOrderID: FMSId) {
-
+async function fetchSamplesAtStepOrder(studyID: FMSId, stepOrderID: number, offset = 0, limit = 10) {
 	const studySettingsByID = selectStudySettingsByID(store.getState())
 
 	// Get the current set of filters and sort order from UX settings for study and step
-	let options = {}
+	let options = { limit } as any
 	const settings = studySettingsByID[studyID]?.stepSettings[stepOrderID]
 	if (settings) {
 		const serializedFilters = settings.filters ? serializeFilterParamsWithDescriptions(settings.filters) : {}
 		const ordering = settings.sortBy ? serializeSortByParams(settings.sortBy) : undefined
-		options = {ordering, ...serializedFilters}
+		const limit = settings.pageSize
+		options = {...options, ordering, limit, ...serializedFilters}
 	}
 
-	return store.dispatch(api.sampleNextStepByStudy.getStudySamplesForStepOrder(studyID, stepOrderID, {...options, limit: 100000}))
+	return store.dispatch(api.sampleNextStepByStudy.getStudySamplesForStepOrder(studyID, stepOrderID, {...options, offset}))
```

Comment:
if offset comes from the props so should the limit (they are expected to go together typically ...) otherwise you are likely to have pages that skip or repeat items ...